rfcs/0000-rfc-process.md

```diff
@@ -0,0 +1,127 @@
+bit of a design process and produce a consensus among the Nix community and
+the core team.
+
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the ecosystem to understand, and implement.  This should get
+into specifics and corner-cases, and include examples of how the feature is
+used.
+
+## When you need to follow this process
+
+You need to follow this process if you intend to make "substantial" changes to
+the Nix ecosystem. What constitutes a "substantial" change is evolving based
+on community norms, but may include the following.
+
+  - Any semantic or syntactic change to the language that is not a bugfix.
+  - Removing language features
+  - Big restructuring of nixpkgs
+  - Introduction of new interfaces or functions
+  - A controversial change
+
+Some changes do not require an RFC:
+
+  - Adding, updating and removing packages in nixpkgs
+  - Additions only likely to be _noticed by_ other developers-of-nix,
+    invisible to users-of-nix.
+
+If you submit a pull request to implement a new feature without going
+through the RFC process, it may be closed with a polite request to
+submit an RFC first.
+
+## What the process is
+
+In short, to get a major feature added to the Nix ecosystem, one should first
+go throught the RFC process in order to improve the likelyhood of inclusion.
+Here are roughly the steps that one would take:
+
+* Fork the RFC repo https://github.com/NixOS/rfcs
+* Copy `0000-template.md` to `rfcs/0000-my-feature.md` (where
+'my-feature' is descriptive. don't assign an RFC number yet).
+* Fill in the RFC
+* Submit a pull request. The pull request is the time to get review of
+the design from the larger community.
+* Build consensus and integrate feedback. RFCs that have broad support
+are much more likely to make progress than those that don't receive any
+comments.
+
+At this point, the person submitting the RFC should find at least one "buddy"
+that will help him bring the RFC to reality. The goal is to improve the
+chances that the RFC is both desired and likely to be implemented.
+
+Whomever merges the RFC should do the following:
+
+* Assign an id, using the PR number of the RFC pull request. (If the RFC
+  has multiple pull requests associated with it, choose one PR number,
+  preferably the minimal one.)
+* Add the file in the `rfcs/` directory.
+* Create a corresponding issue on the appropriate repo (NixOS/nix, NixOS/nixpkgs, ...).
+* Fill in the remaining metadata in the RFC header, including links for
+  the original pull request(s) and the newly created issue.
+* Commit everything.
+
+Once an RFC becomes active then authors may implement it and submit the
+feature as a pull request to the nix or nixpkgs repo. An 'active' is not a
+rubber stamp, and in particular still does not mean the feature will
+ultimately be merged; it does mean that in principle all the major
+stakeholders have agreed to the feature and are amenable to merging it.
+
```

Comment:
Ok, I thought that a graphviz document, with comment would be easier to follow.  There is one thing which remains to be determine in the following, which is the set of person responsible for accepting or rejecting proposals.

```dot
digraph rfc-state {
  /* Draft stage, any newly create RFC is in a draft stage. Provide draft implementation for the project, such as
   *  Nix, Nixpkgs without adding pull request to other repository, but refer to the draft-ed implementation in forked
   *  version of the projects. */
  draft;

  /* Submitted stage, a comity of persons (to be defined) evaluates the feasibility and compare benefits against the
   * drawbacks.  Ideally the peers (buddy, mentors, ...) should have help clear-up the ground for any questions that
   * might be asked by the comity. */
  submitted;

  /* Accepted stage, a comity of persons, agreed with the feasibility and that benefit outweigh the drawback and that a
   * complete implementation would be implemented in the projects. */
  accepted;

  /* Rejected stage, a comity of persons, disagree with the feasibility and/or that the drawbacks outweigh the benefit.
   * The comity should provide a comments explaining the choice. Any reject draft should not be submitted until all 
   * comments are addressed, and not within a period of 3 months (in order to prevent recurrent submissions) */
  rejected;

  /* Abandoned stage, nor the authors nor the peers (buddy, mentor, ...) can think of an alternative to compensate for
   * the rejection comments. */
  abandoned;

  /* Implemented stage, the proposal is available main-stream, and should be the default way of doing. */
  implemented;

  /* Replaced stage, another proposal is implemented and the content of this proposal should be ignored, unless
   * explicitly mentioned by an implemented proposal. */
  replaced;

  draft -> submitted [label="submitted by the author"];
  submitted -> accepted [label="validated by comity"];
  submitted -> rejected [label="rejected by comity"];
  rejected -> draft;

  draft -> abandoned [label="no alternative, or person to pursue"];

  accepted -> implemented [label="all components merged, and the feature is usable"];
  accepted -> replaced [label="better alternative found before the implementation"];
  implemented -> replaced [label="better alternative found after the implementation"];
}
```